### PR TITLE
TimestampRange filter accepts 'now' as bound

### PR DESCRIPTION
**Pull request by tommyblue**
#### :question: What

`TimestampRange` filter accepts `"now"` as time bound (both start or end)

#### :white_check_mark: Checklists

_This section contains a list of checklists for common uses, please delete the checklists that are useless for your current use case (or add another checklist if your use case isn't covered yet)._

- [x] Is there unit/integration test coverage for all new and/or changed functionality added in this PR?
- [ ] Have the changes in this PR been functionally tested?
- [ ] Have new components been added to the related `all.go` files?
- [ ] Have new components been added to the documentation website?
- [x] Has `make gofmt-write` been run on the code?
- [x] Has `make govet` been run on the code? Has the code been fixed accordingly to the output?
- [x] Have the changes been added to the [CHANGELOG.md](./CHANGELOG.md) file?
- [ ] Have the steps in [CONTRIBUTING.md](./CONTRIBUTING.md) been followed to update a Go module?

### CODE REVIEW

File: filter/timestamp_range_test.go [hunk 1]
```diff
@@ -19,14 +20,14 @@ func TestTimestampRange(t *testing.T) {
 			name:      "valid",
 			ts:        1580981641, // 2020-02-06 09:34:01
 			startDate: "2019-02-06 09:34:01",
-			endDate:   "2022-02-06 09:34:01",
+			endDate:   "2122-02-06 09:34:01",
```
Comment (arl) [inline filter/timestamp_range_test.go:23]: Why is this change necessary?

File: filter/timestamp_range_test.go [hunk 2]
```diff
@@ -19,14 +20,14 @@ func TestTimestampRange(t *testing.T) {
...
 			want:      true,
 		},
 		{
 			name:      "past lower bound",
 			ts:        1528277641, // 2018-06-06 09:34:01
 			startDate: "2019-02-06 09:34:01",
-			endDate:   "2022-02-06 09:34:01",
+			endDate:   "2122-02-06 09:34:01",
```
Comment (arl) [inline filter/timestamp_range_test.go:30]: same question

File: filter/timestamp_range_test.go [hunk 3]
```diff
@@ -50,6 +51,20 @@ func TestTimestampRange(t *testing.T) {
...
+			want:      true,
+		},
+		{
+			name:      "end now",
+			ts:        int(time.Now().Unix()),
+			startDate: "2018-02-06 09:34:01",
+			endDate:   "now",
+			want:      false,
+		},
```
Comment (arl) [inline filter/timestamp_range_test.go:67]: would be nice to see a test with `"now"` for both bounds

File: filter/timestamp_range.go [hunk 4]
```diff
@@ -68,6 +66,15 @@ func NewTimestampRange(cfg baker.FilterParams) (baker.Filter, error) {
...
 
+func (c *TimestampRangeConfig) getTime(s string) (time.Time, error) {
+	if s == "now" {
+		return time.Now(), nil
+	}
+
+	const timeLayout = "2006-01-02 15:04:05"
+	return time.Parse(timeLayout, s)
+}
+
```
Comment (arl) [inline filter/timestamp_range.go:77]: Not sure this addition warrants a new function IMHO.
I'm not against a tiny bit of repetition, when it helps readability and doesn't hurt performance where you'd need some.

Also `getTime` has been added on the config struct, so from a readability standpoint it shouldn't be declared in-between `NewTimestampRange` and the `Stats` method.

Why not do things the other way around and call the function `setTimes(start, end string) error`?

**Review by arl — APPROVED** [review]
Nice addition.
Some comments about readability

File: filter/timestamp_range_test.go (hunk 1 above)
Comment (tommyblue) [reply filter/timestamp_range_test.go:23]: it's not, I thought about "now" I then I updated to a date more in the future, but I realize we use `ts` for the test

File: filter/timestamp_range_test.go (hunk 3 above)
Comment (tommyblue) [reply filter/timestamp_range_test.go:67]: done

**Review by arl — APPROVED** [review]
